core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java

```diff
@@ -194,6 +193,64 @@ public Job createJob( JobScheduleRequest scheduleRequest )
     return job;
   }
 
+  /**
+   * Wrapper function around {@link SchedulerOutputPathResolver#resolveOutputFilePath()} calls
+   * {@link #getSchedulerOutputPathResolver(JobScheduleRequest)} to get instance.
+   * @param scheduleRequest
+   * @return
+   */
+  protected String resolveOutputFilePath( JobScheduleRequest scheduleRequest ) {
+    SchedulerOutputPathResolver outputPathResolver = getSchedulerOutputPathResolver( scheduleRequest );
+    return outputPathResolver.resolveOutputFilePath();
+  }
+
+  /**
+   * Wrapper function around {@link IScheduler#createJob(String, Class, Map, IJobTrigger, IBackgroundExecutionStreamProvider)} .
+   * Mainly allowing for different implementation for the underlying input and output streams
+   * through {@link #createIBackgroundExecutionStreamProvider(String, String, JobScheduleRequest)}
+   * @param jobName
+   * @param action
+   * @param jobParams
+   * @param trigger
+   * @param inputFilePath
+   * @param outputFilePath
+   * @param jobScheduleRequest
+   * @return
+   * @throws SchedulerException
+   */
+  protected IJob schedulerCreateJob( String jobName, String action, Map<String, Serializable> jobParams,
+                        IJobTrigger trigger, final String inputFilePath, final String outputFilePath,
+                                     JobScheduleRequest jobScheduleRequest ) throws SchedulerException {
+    return getScheduler().createJob( jobName, action, jobParams, trigger,
+      createIBackgroundExecutionStreamProvider( inputFilePath, outputFilePath, jobScheduleRequest ) );
+  }
+
+  /**
+   * Instantiate of {@link IBackgroundExecutionStreamProvider}.
+   * @param inputFilePath
+   * @param outputFilePath
+   * @param jobScheduleRequest
+   * @return instance that implements IBackgroundExecutionStreamProvider
+   */
+  public IBackgroundExecutionStreamProvider createIBackgroundExecutionStreamProvider( final String inputFilePath, final String outputFilePath, JobScheduleRequest jobScheduleRequest) {
```

Comment:
Unsure about revealing the `I` of the interface name in the method name.
Also, needs to be public?

Reply:
There isn't really a concrete class that I'm interacting with. Yes `org.pentaho.platform.web.http.api.resources.RepositoryFileStreamProvider` implements the interface `IBackgroundExecutionStreamProvider`. 

The interface which the code interacts with from IScheduler.java is 

```
  IJob createJob( String jobName, String actionId, Map<String, Serializable> jobParams, IJobTrigger trigger,
      IBackgroundExecutionStreamProvider outputStreamProvider ) throws SchedulerException;
```

What would you suggest, if any changes?

Reply:
Just a naming convention. Don't think it's usually reflected in method/field names. But I am fine either way.

Reply:
Also, there's the `public` issue, in case you missed.